kordac/tests/HeadingTest.py

```diff
@@ -13,3 +15,159 @@ def __init__(self, *args, **kwargs):
         self.ext = Mock()
         self.ext.processor_info = ProcessorTest.loadProcessorInfo(self)
         self.ext.jinja_templates = {self.processor_name: ProcessorTest.loadJinjaTemplate(self, self.processor_name)}
+
+    #~
+    # Doc Tests
+    #~
+
+    def test_doc_example_basic(self):
+        test_string = self.read_test_file(self.processor_name, 'doc_example_basic_usage.md')
+        blocks = self.to_blocks(test_string)
+
+        self.assertListEqual([True, True, True], [HeadingBlockProcessor(self.ext, self.md.parser).test(blocks, block) for block in blocks], msg='"{}"'.format(test_string))
+
+        converted_test_string = markdown.markdown(test_string, extensions=[self.kordac_extension])
+        expected_string = self.read_test_file(self.processor_name, 'doc_example_basic_usage_expected.html', strip=True)
+        self.assertEqual(expected_string, converted_test_string)
+
+        tree = self.kordac_extension.get_heading_tree()
+        expected_tree = (HeadingNode(title='This is an H1',
```

Comment:
If it is in a file then we even have to write a loader for them too, since they are really python objects.